### PR TITLE
fix(UI): make the task challenges be in the grid view

### LINKED ISSUE #52679: english curriculum grid view
### Is your feature request related to a problem? Please describe.

In general, for projects with no-name lessons, we have been heading in the grid-view direction. Should we do the same for English?

### Describe the solution you'd like

I realise the blocks include "landmark" lessons (dialogues). So, perhaps we want those grid elements to be differently demarcated?

### Describe alternatives you've considered

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mmatsumoto1026**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #52679

This PR was opened to make the task challenges in [A2 English for Developers Curriculum](https://www.freecodecamp.org/learn/a2-english-for-developers/#learn-greetings-in-your-first-day-at-the-office) be in the grid view.

|before|after|
|---|---|
|before|after|

By using gitpod, the code was tested with following environment.

1. Windows 11 Home PC using Chrome, Firefox, Edge, and Brave browser, or, webkit within the Playwright.
1. Android Studio's emulator(Pixel 6 Pro with Android API 34 system image) using Chrome.

### CODE REVIEW

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hi @mmatsumoto1026, this is looking great. The only thing I find a bit confusing is the mixture of what to render and how to render. i.e. the components are mixed together with the logic.

Could you refactor this so what goes into `challengesGrid` and `stepsOrTasks` is a bit cleaner?

e.g. 

```jsx
stepsOrTasks.push(<ComponentNameOfYourChoosing challenge={challengesWithCompleted[i]} .../>)
```

something like that?

Part of the confusion comes from keeping track of what `i` is doing. Ideally, the transformations should all precede the rendering, so that the whole process is

```
// some complicated process to get the data into the right shape..
data.map(thing => {
  if(thing.type === 'dialogSection') return <DialogSection/> else {
  return <ChallengeSection/>
  })
```

Obviously that's a massive oversimplification, and there are lot of ways of implementing that, but I hope that makes sense.

**Comment by mmatsumoto1026** [comment]
Thanks for your review, @ojeytonwilliams.

After reading your comment, I noticed that my first try has broken the structure of grid by placing dialogues and collections of tasks separately.

Therefore, with following your guidance, the code was refactored with using `challengesWithCompleted.map` method in `challenges.tsx` and then adding the className `map-dialogue-wrap` to the dialogue's `li` element.
And also I modified the grid parameters in `intro.css` so that dialogues and tasks could be placed in the same grid.

File: client/src/templates/Introduction/components/challenges.tsx [hunk 1]
```diff
@@ -9,6 +9,8 @@ import GreenNotCompleted from '../../../assets/icons/green-not-completed';
 import GreenPass from '../../../assets/icons/green-pass';
 import { executeGA } from '../../../redux/actions';
 import { ChallengeWithCompletedNode } from '../../../redux/prop-types';
+import { SuperBlocks } from '../../../../../shared/config/superblocks';
+import { challengeTypes } from '../../../../../shared/config/challenge-types';
```
Comment (ojeytonwilliams) [inline client/src/templates/Introduction/components/challenges.tsx:13]:
```suggestion
import { challengeTypes } from '../../../../../shared/config/challenge-types';

const getStepNumber = (dashedName: string) => {
  // dashedName should be in the format 'step-1' or 'task-1'
  const match = dashedName.match(/-(\d+)/);
  return match ? match[1] : '';
};
```
It's a bit verbose, but I think it's clearer as a result.

File: client/src/templates/Introduction/components/challenges.tsx [hunk 2]
```diff
@@ -56,40 +80,51 @@ function Challenges({
...
                   }`}
                 >
-                  <span className='sr-only'>{t('aria.step')}</span>
-                  <span>{i + 1}</span>
+                  <span className='sr-only'>
+                    {challenge.superBlock === SuperBlocks.A2English
+                      ? t('aria.task')
+                      : t('aria.step')}
+                  </span>
+                  <span>{challenge.dashedName.slice(5)}</span>
```
Comment (ojeytonwilliams) [inline client/src/templates/Introduction/components/challenges.tsx:117]:
```suggestion
                  <span>{getStepNumber(challenge.dashedName)}</span>
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
This looks great! It's so much easier to read. 

The only thing I thought was a little confusing was the `slice` to get the step/task number. It's not clear why that's happening - it only makes sense if you know the format of `dashedName`.

**Comment by mmatsumoto1026** [comment]
Thanks for your review, and sorry, it seems that I had not been able to sufficiently consider aproaches other than using the obscure magic number when extracting step/task numbers from the `challenge.dashedName`. 

After testing, the suggested code was pushed.

**Review by ojeytonwilliams — APPROVED** [review]
There's no need to apologize, @mmatsumoto1026. You don't have to think of everything before creating a PR. Personally, I don't expect anyone's code (especially my own!) to be perfect first time - that's why we require multiple reviews.

Thanks again for putting this together. LGTM :+1:

**Review by huyenltnguyen — APPROVED** [review]
@mmatsumoto1026 Thanks for the work!

I've tested locally and verified that 
- English challenges are not displayed in grid view
- The tasks are displayed properly on both dark and light themes
- No visual regressions in other grid-based sections

**Pull request merged**